src/test/synchronizer.test.ts

```diff
@@ -17,57 +17,165 @@ describe("Synchronizer tests", () => {
-    await synchronizer.synchronizeIssue({
-      eventName: "workflow_dispatch",
-      payload: { inputs: { excludeClosed: "true" } },
+  describe("update one issue", () => {
+    let nodeData: NodeData;
+    let ctx: GitHubContext;
+    let issueNumber: number;
+    beforeEach(() => {
+      issueNumber = Math.floor(Math.random() * 100);
+      nodeData = { id: new Date().toDateString(), title: "Update one issue" };
+      ctx = { eventName: "issues", payload: { issue: { node_id: "update_one_issue", number: issueNumber } } };
+      projectKit.fetchProjectData.mockResolvedValue(nodeData);
     });
 
-    expect(logger.notice).toBeCalledWith("Closed issues will NOT be synced.");
-  });
+    test("should invoke using the correct node data", async () => {
+      await synchronizer.synchronizeIssue(ctx);
+      expect(logger.info).toBeCalledWith(`Assigning issue #${issueNumber} to project`);
+      expect(projectKit.assignIssue).toBeCalledWith(ctx.payload.issue, nodeData);
+    });
+
+    test("should return true on correct execution", async () => {
+      expect(await synchronizer.synchronizeIssue(ctx)).toBeTruthy();
+      expect(logger.info).toBeCalledWith(`Assigning issue #${issueNumber} to project`);
+    });
 
-  test("should invoke project.assignIssue", async () => {
-    const issueNumber = Math.floor(Math.random() * 100);
-    await synchronizer.synchronizeIssue({
-      eventName: "issues",
-      payload: { issue: { node_id: "1234321", number: issueNumber } },
+    test("should fetch custom fields data", async () => {
+      const [field, value] = ["a", "B"];
+      await synchronizer.synchronizeIssue({ ...ctx, config: { projectField: { field, value } } });
+      expect(projectKit.fetchProjectFieldNodeValues).toBeCalledWith(nodeData, { field, value });
     });
 
-    logger.info.calledWith(`Assigning issue #${issueNumber} to project`);
-    projectKit.assignIssue.calledWith({ node_id: "1234321", number: issueNumber });
+    test("should assign custom fields", async () => {
+      const nodeValueData = { field: "eee", value: "fff" };
+      projectKit.fetchProjectFieldNodeValues.mockResolvedValue(nodeValueData);
+      const issueCardNodeId = "issue_node_id_example";
+      projectKit.assignIssue.mockResolvedValue(issueCardNodeId);
+      expect(
+        await synchronizer.synchronizeIssue({ ...ctx, config: { projectField: { field: "c", value: "d" } } }),
+      ).toBeTruthy();
+      expect(projectKit.changeIssueStateInProject).toBeCalledWith(issueCardNodeId, nodeData, nodeValueData);
+    });
+
+    test("should throw error while assigning invalid custom field", async () => {
+      projectKit.fetchProjectFieldNodeValues.mockRejectedValue(new Error());
+      await expect(
+        synchronizer.synchronizeIssue({ ...ctx, config: { projectField: { field: "c", value: "d" } } }),
+      ).rejects.toThrow();
+      expect(logger.error).toBeCalledWith("Failed fetching project values.");
```

Comment:
will this error log bring attention of the actiion owner? i mean will they get an email notification or something similar, in case if it started failing? or they will have to be checking the logs to get a sense?

Reply:
In theory you get emails when you github action failed.

I'm thinking of [adding a badge](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge) to the repos that implement it so at least you can see in the readme if the action is failing.

Reply:
I just tested it. I got a notification
![image](https://user-images.githubusercontent.com/8524599/214044779-119be7c0-a9d4-4215-bd9d-ade1449273e3.png)

Reply:
cool, thanks!